modules/sdk-core/src/bitgo/tss/ecdsa/ecdsa.ts

```diff
@@ -295,3 +251,263 @@ export async function encryptNShare(
     encryptedPrivateShare,
   };
 }
+
+/**
+ * Decrypts encrypted n share
+ * @param encryptedNShare - decryptable n share with recipient private gpg key armor and sender public gpg key
+ * @returns N share
+ */
+export async function decryptNShare(encryptedNShare: DecryptableNShare): Promise<NShare> {
+  const privateShare = await readSignedMessage(
+    encryptedNShare.nShare.encryptedPrivateShare,
+    encryptedNShare.senderPublicArmor,
+    encryptedNShare.recipientPrivateArmor
+  );
+
+  const nShare: NShare = {
+    i: encryptedNShare.nShare.i,
+    j: encryptedNShare.nShare.j,
+    y: encryptedNShare.nShare.publicShare.slice(0, 65),
+    u: privateShare,
+    n: encryptedNShare.nShare.publicShare.slice(129),
+    chaincode: encryptedNShare.nShare.publicShare.slice(65, 129),
+  };
+
+  return nShare;
+}
+
+/**
+ * Gets public key from common key chain
+ * @param commonKeyChain - common key chain of ecdsa tss
+ * @returns public key
+ */
+export function getPublicKey(commonKeyChain: string): string {
+  return commonKeyChain.slice(0, 66);
+}
+
+/**
+ * validates signature share record
+ * @param share - signature share record to validate
+ * @param shareLength - share record expected length
+ * @param isExactLength - if this false then share length can be greater.
+ */
+function validateShare(share: SignatureShareRecord, shareLength: number, isExactLength = true): void {
+  const error = (message: string) => new Error(message);
+
+  if (share.share.length < shareLength) {
+    throw error(`Excepted share length to be greater than or equal ${shareLength} but got ${share.share.length}.`);
+  }
+
+  if (isExactLength && share.share.length !== shareLength) {
+    throw error(`Excepted share length to be ${shareLength} but got ${share.share.length}.`);
+  }
+
+  if (share.from === share.to) {
+    throw error(`Key share sender and receiver cannot be the same.`);
+  }
+}
+
+/**
+ * parses K share from signature share record
+ * @param share - signature share record
+ * @returns K Share
+ */
+export function parseKShare(share: SignatureShareRecord): KShare {
+  validateShare(share, 2304);
+
+  return {
+    i: getParticapantIndex(share.to),
+    j: getParticapantIndex(share.from),
+    k: share.share.substring(0, 1536),
+    n: share.share.substring(1536),
+  };
+}
+
+/**
+ * convert K share to signature share record
+ * @param share - K share
+ * @returns signature share record
+ */
+export function convertKShare(share: KShare): SignatureShareRecord {
+  return {
+    to: getParticapantFromIndex(share.i),
+    from: getParticapantFromIndex(share.j),
+    share: share.k + share.n,
+  };
+}
+
+/**
+ * parses A share from signature share record
+ * @param share - signature share record
+ * @returns A Share
+ */
+export function parseAShare(share: SignatureShareRecord): AShare {
+  validateShare(share, 5376, false);
```

Comment:
n may be dynamic, lets check for > 4608

Reply:
the false param is making validateShare to only check for greater.